BitwardenShared/Core/Platform/Utilities/ErrorReportBuilder.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import MachO // dyld
+
+/// A helper object to build error reports to provide detailed error information to share about the
+/// error that occurred.
+///
+struct ErrorReportBuilder {
+    // MARK: Properties
+
+    /// The service used by the application to get info about the app and device it's running on.
+    private let appInfoService: AppInfoService
+
+    // MARK: Initialization
+
+    /// Initialize an `ErrorReportBuilder`.
+    ///
+    /// - Parameter appInfoService: The service used by the application to get info about the app
+    ///     and device it's running on.
+    ///
+    init(appInfoService: AppInfoService) {
+        self.appInfoService = appInfoService
+    }
+
+    // MARK: Methods
+
+    func buildShareErrorLog(for error: Error, callStack: String) -> String {
+        """
+        \(error as NSError)
+        \(error.localizedDescription)
+
+        Stack trace:
+        \(callStack)
+
+        Binary images:
+        \(binaryImageAddresses())
+
+        \(appInfoService.appInfoWithoutCopyrightString)
+        """
+    }
```

Comment:
⛏️ Add docs.
🤔 Could we add a title as the first line of the report log? So when the user shares the file they don't see the `NSError` description in the bottom sheet title and instead they can see a more user friendly title.

Reply:
Yeah makes sense, we can leave it as is, thanks!